packages/markdown/loader.js

```diff
@@ -438,32 +460,104 @@ module.exports = async function demoLoader() {
           demos[demoName].relativeModules = {};
         }
 
+        const addedModulesRelativeToModulePath = new Set();
         await Promise.all(
           Array.from(relativeModules.get(demoName)).map(async ([relativeModuleID, variants]) => {
-            let raw = '';
-            try {
-              raw = await fs.readFile(path.join(path.dirname(moduleFilepath), relativeModuleID), {
-                encoding: 'utf8',
-              });
-            } catch {
-              throw new Error(
-                `Could not find a module for the relative import "${relativeModuleID}" in the demo "${demoName}"`,
+            for (const variant of variants) {
+              let raw = '';
+              const relativeModuleFilePath = path.join(
+                path.dirname(moduleFilepath),
+                relativeModuleID,
               );
```

Comment:
From what I understand, this `relativeModuleFilePath` is the file you are processing.

It looks like you process multiple time the same file, because if `A.tsx` and `B.tsx` both import `common.types.ts`

Is it necessary for some data creation, or can we do an early return for files that have already been processed?

Reply:
You are correct, I am checking the `addedModulesRelativeToModulePath` to make sure I am not adding duplicates, but for sure, we can return early if the file has been processed and go to the next one 👍 Nice catch!